dkist/wcs/models.py

```diff
@@ -0,0 +1,340 @@
+from typing import Any, Union, Iterable
+
+import numpy as np
+
+try:
+    from numpy.typing import ArrrayLike  # NOQA
+except ImportError:
+    ArrayLike = Any
+
+import astropy.modeling.models as m
+import astropy.units as u
+from astropy.modeling import CompoundModel, Model, Parameter
+
+__all__ = ['CoupledCompoundModel',
+           'InverseVaryingCelestialTransform',
+           'VaryingCelestialTransform',
+           'BaseVaryingCelestialTransform']
+
+
+def generate_celestial_transform(crpix: Union[Iterable[float], u.Quantity],
+                                 cdelt: Union[Iterable[float], u.Quantity],
+                                 pc: Union[ArrayLike, u.Quantity],
+                                 crval: Union[Iterable[float], u.Quantity],
+                                 lon_pole: Union[float, u.Quantity] = None,
+                                 projection: Model = m.Pix2Sky_TAN()) -> CompoundModel:
+    """
+    Create a simple celestial transform from FITS like parameters.
+
+    Supports unitful or unitless parameters, but if any parameters have units
```

Comment:
Based on comments in the test it seems like a unitless set of inputs will be interpreted as degrees. Probably worth writing that down up here.